### DISCUSSION TITLE
Loader Versoid

### DISCUSSION THREAD

**Discussion by Medaturd76** [FabricMC/fabric-api#1594] [Player support]
Go [here](https://github.com/microsoft/vscode-java-pack/issues/687) to see the issue

**Comment by warjort** [comment]
Try changing your gradle properties to be the following from the example mod:
https://github.com/FabricMC/fabric-example-mod/blob/1.17/gradle.properties

In your config, you appear to be specifiying a 1.17.1 version of fabric api, but with minecraft 1.17
https://www.curseforge.com/minecraft/mc-mods/fabric-api/files/3377591

**Reply by Medaturd76** [reply]
tryed. Its the fabric api. its on version 1.17.x and I need it on 1.17

**Comment by warjort** [comment]
According to curseforge, the last version of fabric api for 1.17 was 0.36.0+1.17
https://www.curseforge.com/minecraft/mc-mods/fabric-api/files

**Comment by warjort** [comment]
That version was using yarn version +build.6
https://github.com/FabricMC/fabric/commit/a722d8c028b5de60fb4e40d2552f314110311160#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7

**Comment by Medaturd76** [comment]
nope

**Comment by Medaturd76** [comment]
Where are the fires for the minecraft mods located?

**Reply by warjort** [reply]
In .gradle/caches of your home directory  (note the . in front of gradle) 
unless they need to be remapped (e.g. if you use access wideners) then they are in 
your project's .gradle/loom-cache

**Reply by Medaturd76** [reply]
no, I mean in like the .minecraft folder in AppData or something?

**Comment by warjort** [comment]
The .minecraft folder in AppData is not used in mod development, or is this a totally new question for this thread?

Assuming it is not a totally new question, when you do runclient or runserver
there is a run folder created in your project. This has a mods subfolder, but it is usually empty.
Instead, the mods are loaded from the places I said above based on what dependencies you have in your gradle project.

**Reply by Medaturd76** [reply]
No, I mean the fabric api mod when u run the minecraft client debug thing is for version 1.17.X, I need it on 1.17 . the way you would usualy do this for regular minecraft is go to C:\Users\(User)\AppData\Roaming\.minecraft\mods. Where is the mods folder for this minecraft or how do I change the version of fabric api?

**Comment by warjort** [comment]
As I said above, you change it by changing the dependencies in the gradle project.

Here is a link to a fork I have of 1.17 (not 1.17.1) fabric-example mod
https://github.com/warjort/fabric-example-mod

In the build.gradle you will see the dependency:
````
modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
````

The $ means it is using a property. In this case it is defined in the gradle.properties.
https://github.com/warjort/fabric-example-mod/blob/1.17/gradle.properties
````
fabric_version=0.36.0+1.17
````
This is the version of fabric-api you want for 1.17 and I posted it earlier.
But you said you already tried it and it didn't work, but didn't provide any further information on what didn't work.
I can tell you that fork worked for me, when I used it for some testing a few weeks ago.

**Reply by Medaturd76** [reply]
I did set it to that and didn't work

**Reply by Medaturd76** [reply]
like i said, Fabric api si on version 1.17.X, I need in on 1.17

**Comment by warjort** [comment]
We are not psychic. :-)

I just retested that fork I posted and it works fine.

These are the steps I did:
````
git clone https://github.com/warjort/fabric-example-mod
````
This checkouts my fork of the example mod

Build the project and run minecraft in client mode
````
cd fabric-example-mod
./gradlew runclient
````

This is the output I got from the start of minecraft loading.
````
> Task :runClient
[20:03:44] [main/INFO] (FabricLoader) Loading for game Minecraft 1.17
[20:03:44] [main/INFO] (Fabric|Loader) [FabricLoader] Loading 48 mods:
        - fabric@0.36.0+1.17
        - fabricloader@0.11.6
<snip/>
        - java@16
        - minecraft@1.17
        - modid@1.0.0
````

Now you do the same and tell me what is different/doesn't work.

**Reply by Medaturd76** [reply]
was the command ran in an windows shell terminal or cmd?

**Reply by warjort** [reply]
cygwin (unix emulation for windows)
for windows you run gradlew.bat
you also need a copy of git if you don't have it

Or you could just do the same operations in your ide.

**Reply by Medaturd76** [reply]
oh. it worked! thx

**Reply by Medaturd76** [reply]
wait, it worked but my mod items and blocks have no texture, the item does 1/2 things that its suppost to do

**Comment by warjort** [comment]
That's not fabric api's problem. 
You have some debugging to do. :-)

**Reply by Medaturd76** [reply]
But it does not say there are any errors in the item file, just in the block

**Comment by warjort** [comment]
Start a new discussion for a new problem. This thread is already long.
And you need to post more than "it doesn't work" to get any help.

**Reply by Medaturd76** [reply]
k, thx agen
